server/mqtt.go

```diff
@@ -2416,21 +2487,55 @@ func (as *mqttAccountSessionManager) processSubs(sess *mqttSession, c *client,
 			}
 		}
 
-		// Load retained messages.
+		// Find retained messages.
 		if fromSubProto {
-			if err := loadRMS([]byte(f.filter)); err != nil {
+			addRMSubjects := func(subject string) error {
+				sub := &subscription{
+					client:  c,
+					subject: []byte(subject),
+					sid:     []byte(subject),
+				}
+				if err := c.addShadowSubscriptions(acc, sub, false); err != nil {
+					return err
+				}
+
+				subs := []*subscription{sub}
+				if len(sub.shadow) > 0 {
+					subs = append(subs, sub.shadow...)
+				}
+				for _, sub := range subs {
+					subject := string(sub.subject)
```

Comment:
nit: Derek introduced bytesToString() which avoid mem alloc. I believe that in this case, since we are not going to store/etc.. the subject and use it just for the reverse match, it could be safe to do so.